receiver/hostmetricsreceiver/hostmetrics_receiver_test.go

```diff
@@ -133,3 +134,44 @@ func assertMetricDataAndGetMetricsSlice(t *testing.T, metrics pdata.Metrics) pda
 	assert.Equal(t, 1, ilms.Len())
 	return ilms.At(0).Metrics()
 }
+
+func benchmarkScrapeMetrics(b *testing.B, cfg *Config) {
+	sink := &exportertest.SinkMetricsExporter{}
+
+	factories := map[string]internal.Factory{
```

Comment:
Might be worth extracting factories to `Benchmark_ScrapeMetricsAll` as well, since they match the config?

Reply:
I moved this to a global variable shared by all the tests. We may as well always inject all factories; only the ones that config is provided for will actually be used.